### PR TITLE
Main solution E2E

### PR DESCRIPTION
**Pull request by android-dev-lxl**

### CODE REVIEW

**Review by johnshea — CHANGES_REQUESTED** [review]
The UI on API 33 needs to be adjusted. It changed and no longer matches the pre e2e UI.

Screenshot 2023-08-21 at 9 54 51 AM

File: build.gradle.kts [hunk 1]
```diff
@@ -16,12 +16,12 @@
 
 buildscript {
     extra.apply {
-        set("compose_compiler_version", "1.4.7")
+        set("compose_compiler_version", "1.5.1")
```
Comment (johnshea) [inline build.gradle.kts:19]: Update `kotlinCompilerExtensionVersion`

File: build.gradle.kts [hunk 2]
```diff
@@ -16,12 +16,12 @@
...
     }
 }
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id("com.android.application") version "8.0.2" apply false
-    id("com.android.library") version "8.0.2" apply false
-    id("org.jetbrains.kotlin.android") version "1.8.21" apply false
+    id("com.android.application") version "8.1.0" apply false
```
Comment (johnshea) [inline build.gradle.kts:24]: Update AGP

File: app/src/main/java/com/example/tiptime/MainActivity.kt [hunk 3]
```diff
@@ -83,19 +83,18 @@ fun TipTimeLayout() {
...
+            .padding(horizontal = 40.dp)
             .verticalScroll(rememberScrollState()),
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.Center
     ) {
         Text(
             text = stringResource(R.string.calculate_tip),
             modifier = Modifier
-                .padding(bottom = 16.dp)
+                .padding(bottom = 16.dp, top = 40.dp)
```
Comment (johnshea) [inline app/src/main/java/com/example/tiptime/MainActivity.kt:97]: Could we add 40.dp to the dimens.xml? It is used in multiple spots.

File: app/src/main/java/com/example/tiptime/MainActivity.kt
Comment (johnshea) [inline app/src/main/java/com/example/tiptime/MainActivity.kt:1]: Could we add 32.dp to the dimens.xml? Used in multiple spots.

**Review by johnshea — CHANGES_REQUESTED** [review]
Navigation bar is not translucent (Pixel 6 API 33) but should be.

Screenshot 2023-08-25 at 10 59 27 AM

API 24 - Navigation bar is not translucent but should be. (Note - I added additional `Text` to test.)

Screenshot 2023-08-25 at 11 29 26 AM

API 26 - Navigation bar is not translucent but should be. (Note - I added additional `Text` to test.)
Screenshot 2023-08-25 at 11 29 58 AM

**Review by johnshea — CHANGES_REQUESTED** [review]
Issue 1

UI components are not positioned the same for "System Navigation - Gesture" vs "System Navigation - 3 button" (Pixel 6 API 33)

Note - Red lines drawn to show how UI components are not positioned the same. Lines are not part of the actual UI.

Screenshot 2023-08-30 at 6 18 09 PM

-----

Issue 2

UI scrolls vertically through the status bar.

Is this the intended behavior?

Screenshot 2023-08-30 at 6 29 07 PM

File: app/src/main/java/com/example/tiptime/MainActivity.kt (hunk 3 above)
Comment (android-dev-lxl) [reply app/src/main/java/com/example/tiptime/MainActivity.kt:97]: We do not have `dimens.xml` for this project.

File: app/src/main/java/com/example/tiptime/MainActivity.kt
Comment (android-dev-lxl) [reply app/src/main/java/com/example/tiptime/MainActivity.kt:1]: Same as above, me and SME decided that is too early to introduce dimens.xml

**Comment by android-dev-lxl** [comment]
> Navigation bar is not translucent (Pixel 6 API 33) but should be.
> 
> Screenshot 2023-08-25 at 10 59 27 AM
> API 24 - Navigation bar is not translucent but should be. (Note - I added additional `Text` to test.)
> 
> Screenshot 2023-08-25 at 11 29 26 AM
> API 26 - Navigation bar is not translucent but should be. (Note - I added additional `Text` to test.) Screenshot 2023-08-25 at 11 29 58 AM



Issue 1 - Is not an issue and is expected. We can chat about this, if needed. 
Issue 2 - Fixed

**Comment by android-dev-lxl** [comment]
> Issue 1
> 
> UI components are not positioned the same for "System Navigation - Gesture" vs "System Navigation - 3 button" (Pixel 6 API 33)
> 
> Note - Red lines drawn to show how UI components are not positioned the same. Lines are not part of the actual UI.
> 
> Screenshot 2023-08-30 at 6 18 09 PM
> Issue 2
> 
> UI scrolls vertically through the status bar.
> 
> Is this the intended behavior?
> 
> Screenshot 2023-08-30 at 6 29 07 PM

Issue 1 - Is not an issue and is expected. We can chat about this, if needed.
Issue 2 - Fixed

**Review by johnshea — APPROVED** [review]
lgtm

one nit - remove comment in MainActivity.kt as the code it refers to was removed.

File: app/src/main/java/com/example/tiptime/ui/theme/Theme.kt [hunk 4]
```diff
@@ -17,6 +17,7 @@ package com.example.tiptime.ui.theme
 
 import android.app.Activity
```
Comment (johnshea) [inline app/src/main/java/com/example/tiptime/ui/theme/Theme.kt:18]: Remove unused imports

**Review by johnshea — APPROVED** [review]
Approved but unused imports need to be removed.

**Pull request merged**